src/main/java/com/aws/greengrass/componentmanager/ComponentServiceHelper.java

```diff
@@ -65,8 +72,24 @@ ResolvedComponentVersion resolveComponentVersion(String componentName, Semver lo
                 .componentCandidates(Collections.singletonList(candidate)).build();
 
         ResolveComponentCandidatesResponse result;
-        try {
-            result = clientFactory.getGreengrassV2DataClient().resolveComponentCandidates(request);
+
+        RetryUtils.RetryConfig clientExceptionRetryConfig =
+                RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofSeconds(30))
+                        .maxRetryInterval(Duration.ofSeconds(30)).maxAttempt(CLIENT_RETRY_COUNT)
+                        .retryableExceptions(Arrays.asList(DeviceConfigurationException.class)).build();
+
+        try (GreengrassV2DataClient greengrasV2DataClient = RetryUtils.runWithRetry(clientExceptionRetryConfig, () -> {
+            GreengrassV2DataClient client = clientFactory.getGreengrassV2DataClient();
```

Comment:
would it be better to keep and deprecate this API and add a new one that `throws DeviceConfigurationException`?

I'm not sure what a useful name for that API would be though

Reply:
Was wondering if we should add a new API. What's our process for deprecation APIs without breaking folks using them?

Reply:
Spoke offline. Will add @Deprecated and create a new API.